lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -3056,10 +3056,11 @@ ParseResult FIRStmtParser::parseMem(unsigned memIndent) {
       if (sym)
         break;
     }
-  result = builder.create<MemOp>(resultTypes, readLatency, writeLatency, depth,
-                                 ruw, builder.getArrayAttr(resultNames), id,
-                                 inferNameKind(id), annotations.first,
-                                 annotations.second, sym, IntegerAttr());
+  result = builder.create<MemOp>(
+      resultTypes, readLatency, writeLatency, depth, ruw,
+      builder.getArrayAttr(resultNames), id, inferNameKind(id),
+      annotations.first, annotations.second,
+      sym ? InnerSymAttr::get(sym) : InnerSymAttr(), IntegerAttr());
```

Comment:
Is there a reason to not simplify this to use the StringAttr builder, which contains the conditional creation of the InnerSymAttr for us?

Same for the other instances of this pattern.

Reply:
This particular version of the builder does not support the `StringAttr`. Basically, this is using the auto genreated builder and not the extra builders specified in the `FIRRTLDeclarations.td`.